src/test/pybind/test_rados.py

```diff
@@ -444,6 +444,17 @@ def test_snap_read(self):
         self.ioctx.remove_snap("snap1")
         self.ioctx.remove_object("inhead")
 
+    def test_assert_exists(self):
+        with WriteOpCtx() as write_op:
+            write_op.new(0)
+            write_op.assert_exists()
+            self.ioctx.operate_write_op(write_op, 'abc')
+            eq(self.ioctx.read('abc'), b'')
+            write_op.remove()
+            self.ioctx.operate_write_op(write_op, "abc")
+            with assert_raises(ObjectNotFound):
+                self.ioctx.read('abc')
```

Comment:
@zjcmszh could you help understand how this is related to a negative case of `write_op.assert_exists()`? i mean, even if `write_op.assert_exists()` does not fail, `self.ioctx.read('abc')` can trigger the exception as well.

Reply:
@tchaikov  thank you for your suggestion, I'll figure out how to fix it

Reply:
@zjcmszh ping?